subsys/logging/Kconfig

```diff
@@ -223,6 +225,13 @@ config LOG_CMDS
 	depends on SHELL
 	default y if SHELL
 
+config LOG_FRONTEND
+	bool "Enable frontend"
+	default n
```

Comment:
i think that it should `select LOG_IMMEDIATE` since it is a form of immediate logging and it should take care of removing portions of code and data and remove some kconfig options related to deferred logging. Additionally, it will disable string duplication which also does not make sense in immediate logging.